syntax = "proto3";
package article;

service ArticleService {
  rpc ReadArticleList(ReadArticleListRequest)
      returns (ReadArticleListResponse) {}
  rpc ReadArticleListBySearchterm(ReadArticleListBySearchtermRequest)
      returns (ReadArticleListBySearchtermResponse) {}
  rpc ReadArticleListBySite(ReadArticleListBySiteRequest) returns (ReadArticleListBySiteResponse) {}

}

message ArticleMessage {
  string name = 1;
  string site = 2;
  string url = 3;
  string language = 4;
  string scrape_date = 5;
  optional string submission_date = 6;
  repeated string tags_and_categories = 7;
}

// ##### Request #####

message ReadArticleListRequest {}
message ReadArticleListBySearchtermRequest { string search_term = 1; }
message ReadArticleListBySiteRequest { string site = 1; }

// ##### Response #####

message ReadArticleListResponse {
  repeated ArticleMessage articles = 1;
  string msg = 2;
}
message ReadArticleListBySearchtermResponse {
  repeated ArticleMessage articles = 1;
  string msg = 2;
}
message ReadArticleListBySiteResponse {
  repeated ArticleMessage articles = 1;
  string msg = 2;
}
